### PR TITLE
test `advance`

### PR DESCRIPTION
**Pull request by AABur**
RE: #232 

Tests for `advance`

Additional: implemented `pytest-mock` instead of `unittest.mock`

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -13,6 +13,7 @@ install_command =
 deps =
   pytest
   pytest-xdist
+  pytest-mock
```
Comment (PawelLipski) [inline tox.ini:16]: If it comes to rationale for the change around mocks... this is for the sake of making the tests more consistent with pytest conventions right? Or does it have some other positive side effects?

**Comment by PawelLipski** [comment]
Also! what else is to be done on this PR before it's no longer a draft?

File: tox.ini (hunk 1 above)
Comment (AABur) [reply tox.ini:16]: In short, `pytest-mock` allows to write very concise tests and reduces a lot of boilerplate.
It also provides the very useful `mocker.spy()` and `mocker.stub()`  features, something doable with `unittest.mock` but requires lots of code. For more info read [pytest-mock documentation](https://pytest-mock.readthedocs.io/en/latest/)

**Comment by AABur** [comment]
> Also! what else is to be done on this PR before it's no longer a draft?

I made two commits on purpose to highlight the changes made by using `pytest-mock`. You can see them in this commit `tests with pytest-mock`

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**